tests/integration/security/authorization_test.go

```diff
@@ -906,54 +831,109 @@ func TestAuthorization_TCP(t *testing.T) {
 				}
 			}
+			ns := apps.Namespace1
+			ns2 := apps.Namespace2
+			a := apps.A.Match(echo.Namespace(ns.Name()))
+			b := apps.B.Match(echo.Namespace(ns.Name()))
+			c := apps.C.Match(echo.Namespace(ns.Name()))
+			cInNS2 := apps.C.Match(echo.Namespace(ns2.Name()))
+			d := apps.D.Match(echo.Namespace(ns.Name()))
+			e := apps.E.Match(echo.Namespace(ns.Name()))
+			t.NewSubTest("").
+				Run(func(t framework.TestContext) {
+					policy := tmpl.EvaluateAllOrFail(t, map[string]string{
+						"Namespace":  ns.Name(),
+						"Namespace2": ns2.Name(),
+						"dst0":       b[0].Config().Service,
+						"dst1":       c[0].Config().Service,
+						"dst2":       d[0].Config().Service,
+						"dst4":       a[0].Config().Service,
+						"src0":       a[0].Config().Service,
+						"src1":       b[0].Config().Service,
+					}, file.AsStringOrFail(t, "testdata/authz/v1beta1-tcp.yaml.tmpl"))
+					t.Config().ApplyYAMLOrFail(t, "", policy...)
+					cases := []rbacUtil.TestCase{
+						// The policy on workload b denies request with path "/data" to port 8091:
+						// - request to port http-8091 should be denied because both path and port are matched.
+						// - request to port http-8092 should be allowed because the port is not matched.
+						// - request to port tcp-8093 should be allowed because the port is not matched.
+						newTestCase(a, b, "http-8091", false, scheme.HTTP),
+						newTestCase(a, b, "http-8092", true, scheme.HTTP),
+						newTestCase(a, b, "tcp-8093", true, scheme.TCP),
+
+						// The policy on workload c denies request to port 8091:
+						// - request to port http-8091 should be denied because the port is matched.
+						// - request to http port 8092 should be allowed because the port is not matched.
+						// - request to tcp port 8093 should be allowed because the port is not matched.
+						// - request from b to tcp port 8093 should be allowed by default.
+						// - request from b to tcp port 8094 should be denied because the principal is matched.
+						// - request from cInNS2 to tcp port 8093 should be denied because the namespace is matched.
+						// - request from cInNS2 to tcp port 8094 should be allowed by default.
+						newTestCase(a, c, "http-8091", false, scheme.HTTP),
+						newTestCase(a, c, "http-8092", true, scheme.HTTP),
+						newTestCase(a, c, "tcp-8093", true, scheme.TCP),
+						newTestCase(b, c, "tcp-8093", true, scheme.TCP),
+						newTestCase(b, c, "tcp-8094", false, scheme.TCP),
+						newTestCase(cInNS2, c, "tcp-8093", false, scheme.TCP),
+						newTestCase(cInNS2, c, "tcp-8094", true, scheme.TCP),
+
+						// The policy on workload d denies request from service account a and workloads in namespace 2:
+						// - request from a to d should be denied because it has service account a.
+						// - request from b to d should be allowed.
+						// - request from e to d should be allowed.
+						// - request from cInNS2 to e should be allowed because there is no policy on b.
```

Comment:
`... no policy on b.`, should this be `no policy on e`?

Reply:
I have updated the workloads for this test. Now the comments above each test case should not be changed.